modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts

```diff
@@ -566,6 +576,35 @@ describe('browser.tinymce.core.html.SchemaTest', () => {
 
       assert.hasAnyKeys(schema.getNonEmptyElements(), [ 'foo-bar', 'FOO-BAR', 'bar-foo', 'BAR-FOO' ]);
     });
+
+    it('TINY-10139: Check block elements', () => {
+      const schema = Schema({
+        custom_elements: 'foo,bar'
+      });
+      checkElement('p', (el) => schema.isBlock(SugarNode.name(el)), true);
+      checkElement('h1', (el) => schema.isBlock(SugarNode.name(el)), true);
+      checkElement('table', (el) => schema.isBlock(SugarNode.name(el)), true);
+      checkElement('span', (el) => schema.isBlock(SugarNode.name(el)), false);
+      checkElement('b', (el) => schema.isBlock(SugarNode.name(el)), false);
+      checkText((el) => schema.isBlock(SugarNode.name(el)));
+
+      checkElement('foo', (el) => schema.isBlock(SugarNode.name(el)), true);
+      checkElement('bar', (el) => schema.isBlock(SugarNode.name(el)), true);
+    });
+
+    it('TINY-10139: Check inline elements', () => {
+      const schema = Schema({
+        custom_elements: '~foo,~bar'
+      });
+      checkElement('b', (el) => schema.isInline(SugarNode.name(el)), true);
+      checkElement('span', (el) => schema.isInline(SugarNode.name(el)), true);
+      checkElement('p', (el) => schema.isInline(SugarNode.name(el)), false);
+      checkElement('h1', (el) => schema.isInline(SugarNode.name(el)), false);
+      checkText((el) => schema.isInline(SugarNode.name(el)));
```

Comment:
You can move these to separate `context('isInline')`

Reply:
look at the comment here: https://github.com/tinymce/tinymce/pull/9040#discussion_r1338172502